assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -717,27 +707,33 @@ const baseActions = {
 
 const baseControls = {};
 
-function baseReducer( state, { type, payload } ) {
-	switch ( type ) {
-		case MATCHING_ACCOUNT_PROPERTY:
-			return { ...state, ...payload };
-		case SET_HAS_MISMATCHED_TAG:
-			return {
-				...state,
-				moduleData: {
-					...state.moduleData,
-					hasMismatchedTag: payload.hasMismatchedTag,
-				},
-			};
-		case SET_IS_WEBDATASTREAM_AVAILABLE:
-			return {
-				...state,
-				isWebDataStreamAvailable: payload.isWebDataStreamAvailable,
+const baseReducer = createReducer( ( state, action ) => {
+	switch ( action.type ) {
+		case MATCHING_ACCOUNT_PROPERTY: {
+			const { isMatchingAccountProperty } = action.payload;
+			state.isMatchingAccountProperty = isMatchingAccountProperty;
+			break;
```

Comment:
It would be slightly less repetitive to keep the destructure of `{ type, payload }` in the argument list and then not destructure from `action.payload` in the case blocks. For example the first case would look like this:

```suggestion
const baseReducer = createReducer( ( state, { type, payload } ) => {
	switch ( type ) {
		case MATCHING_ACCOUNT_PROPERTY: {
			state.isMatchingAccountProperty = payload.isMatchingAccountProperty;
			break;
```

> [!NOTE]
> It would also be nice to retain the destructuring of `{ type, payload }` in the other reducers where it was previously, to reduce noise in the PR.